File: web/src/main/java/by/sorface/sso/web/services/providers/socials/GoogleOAuth2UserDatabaseProvider.java
package by.sorface.sso.web.services.providers.socials;

import by.sorface.sso.web.dao.models.UserEntity;
import by.sorface.sso.web.records.principals.SfPrincipal;
import by.sorface.sso.web.records.socialusers.GoogleOAuth2User;
import by.sorface.sso.web.services.providers.AbstractOAuth2UserDatabaseProvider;
import by.sorface.sso.web.services.users.social.SocialOAuth2UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.core.convert.converter.Converter;
import org.springframework.security.oauth2.core.user.OAuth2User;
import org.springframework.stereotype.Service;

@Service
public class GoogleOAuth2UserDatabaseProvider extends AbstractOAuth2UserDatabaseProvider<GoogleOAuth2User> {

    @Autowired
    public GoogleOAuth2UserDatabaseProvider(final SocialOAuth2UserService<GoogleOAuth2User> oAuth2UserSocialOAuth2UserService,
                                            @Qualifier("defaultSfUserMapper") final Converter<UserEntity, SfPrincipal> principalConverter) {
        super(oAuth2UserSocialOAuth2UserService, principalConverter);
    }

    @Override
    protected GoogleOAuth2User buildOAuth2User(OAuth2User oAuth2User) {
        return GoogleOAuth2User.from(oAuth2User);
    }

}

File: web/src/main/kotlin/by/sorface/passport/web/services/users/providers/socials/GoogleOAuth2UserDatabaseProvider.kt
package by.sorface.passport.web.services.users.providers.socials

import by.sorface.passport.web.converters.PrincipalConverter
import by.sorface.passport.web.converters.socialusers.GoogleOAuth2UserConverter
import by.sorface.passport.web.records.socialusers.GoogleOAuth2User
import by.sorface.passport.web.services.users.providers.AbstractOAuth2UserDatabaseProvider
import by.sorface.passport.web.services.users.social.GoogleOAuth2UserService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class GoogleOAuth2UserDatabaseProvider @Autowired constructor(
    googleOAuth2UserService: GoogleOAuth2UserService,
    principalConverter: PrincipalConverter,
    googleOAuth2UserConverter: GoogleOAuth2UserConverter
) : AbstractOAuth2UserDatabaseProvider<GoogleOAuth2User>(googleOAuth2UserService, principalConverter, googleOAuth2UserConverter)
